From 738e40b5734642ca6a56d2e8842ee77a820e3d2c Mon Sep 17 00:00:00 2001
From: "Jason M. Bills" <jason.m.bills@intel.com>
Date: Tue, 31 Oct 2023 12:16:03 -0700
Subject: [PATCH] Increase crashdump Task timeout

On some systems, crashdump can take longer than 5 minutes resulting in
the Task timing out instead of completing.

This increases the timeout to 30 minutes to make sure longer crashdumps
have a good amount of time to complete and show correctly in the Task.

Tested:
Confirmed that a 10-minute crashdump Task shows as "Completed" instead
of "Cancelled".

Signed-off-by: Jason M. Bills <jason.m.bills@intel.com>
---
 redfish-core/lib/log_services.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/redfish-core/lib/log_services.hpp b/redfish-core/lib/log_services.hpp
index 54284cfe..791dbcb6 100644
--- a/redfish-core/lib/log_services.hpp
+++ b/redfish-core/lib/log_services.hpp
@@ -3707,7 +3707,7 @@ inline void requestRoutesCrashdumpCollect(App& app)
             },
                 taskMatchStr);
 
-            task->startTimer(std::chrono::minutes(5));
+            task->startTimer(std::chrono::minutes(30));
             task->populateResp(asyncResp->res);
             task->payload.emplace(std::move(payload));
         };
-- 
2.25.1

